mlos_bench/mlos_bench/optimizers/grid_search_optimizer.py

```diff
@@ -0,0 +1,188 @@
+        """
+        Gets the set of pending configs in this grid search optimizer.
+
+        Returns
+        -------
+        Iterable[Dict[str, TunableValue]]
+        """
+        # See NOTEs above.
+        return (dict(zip(self._config_keys, config)) for config in self._pending_configs.keys())
+
+    @property
+    def suggested_configs(self) -> Iterable[Dict[str, TunableValue]]:
+        """
+        Gets the set of configs that have been suggested but not yet registered.
+
+        Returns
+        -------
+        Iterable[Dict[str, TunableValue]]
+        """
+        # See NOTEs above.
+        return (dict(zip(self._config_keys, config)) for config in self._suggested_configs)
+
+    def bulk_register(self, configs: Sequence[dict], scores: Sequence[Optional[float]],
+                      status: Optional[Sequence[Status]] = None, is_warm_up: bool = False) -> bool:
+        if not super().bulk_register(configs, scores, status, is_warm_up):
+            return False
+        if status is None:
+            status = [Status.SUCCEEDED] * len(configs)
+        for (params, score, trial_status) in zip(configs, scores, status):
+            tunables = self._tunables.copy().assign(params)
+            self.register(tunables, trial_status, None if score is None else float(score))
+            if is_warm_up:
+                # Do not advance the iteration counter during warm-up.
+                self._iter -= 1
+        if _LOG.isEnabledFor(logging.DEBUG):
+            (score, _) = self.get_best_observation()
+            _LOG.debug("%s end: %s = %s", "Warm-up" if is_warm_up else "Update", self.target, score)
+        return True
+
+    def suggest(self) -> TunableGroups:
+        """
+        Generate the next grid search suggestion.
+        """
+        tunables = self._tunables.copy()
+        if self._start_with_defaults:
+            _LOG.info("Use default values for the first trial")
+            self._start_with_defaults = False
+            tunables = tunables.restore_defaults()
+            # Need to index based on ConfigSpace dict ordering.
+            default_config = dict(self.config_space.get_default_configuration())
+            assert tunables.get_param_values() == default_config
+            # Move the default from the pending to the suggested set.
+            default_config_values = tuple(default_config.values())
+            del self._pending_configs[default_config_values]
+            self._suggested_configs.add(default_config_values)
+        else:
+            # Select the first item from the pending configs.
+            if not self._pending_configs and self._iter <= self._max_iter:
+                _LOG.info("No more pending configs to suggest. Restarting grid.")
+                self._config_keys, self._pending_configs = self._get_grid()
+            try:
+                next_config_values = next(iter(self._pending_configs.keys()))
+            except StopIteration as exc:
```

Comment:
maybe
```suggestion
                next_config_values = self._pending_configs.popitem()[0]
            except KeyError as exc:
```
and remove that `del` below?